src/app/content/components/Topbar/index.tsx

```diff
@@ -35,191 +38,327 @@ interface Props {
+            <Styled.CloseIcon />
+          </Styled.CloseButtonNew>
+        }
+        {
+          state.query && !newButtonEnabled && <Styled.CloseButton
+            type='button'
+            aria-label='clear search'
+            onClick={onSearchClear}
+            formSubmitted={state.formSubmitted}
+            data-testid='mobile-clear-search'
+          />
+        }
+      </Styled.SearchInputWrapper>
+      {children}
+    </Styled.MobileSearchContainer>
+  </Styled.MobileSearchWrapper>
+
+  );
+}
+
+// Effectively a conditionally executed hook
+function AltSCycler({hasSearchResults}: {hasSearchResults: boolean}) {
+  const isMobile = useMatchMobileQuery();
+  const cycleSearchRegions = React.useCallback(
+    // Only in desktop layout
+    () => {
+      if (isMobile) {
+        return;
+      }
+      const targets = [
+        '[class*="SearchInputWrapper"] input',
+        '[class*="SearchResultsBar"]',
+        'main',
+      ].map((q) => document?.querySelector(q));
+
+      // Determine which region we are in (if any)
+      const currentSectionIndex = targets.findIndex((el) => el?.contains(document?.activeElement!));
+
+      // If not in any, go to search input
+      if (currentSectionIndex < 0) {
+        (targets[0] as HTMLElement).focus();
+        return;
+      }
+      // If there are no search results, toggle between search input and main content
+      if (!hasSearchResults) {
+        (targets[currentSectionIndex === 0 ? 2 : 0] as HTMLElement).focus();
+        return;
+      }
+      const nextSectionIndex = (currentSectionIndex + 1) % targets.length;
 
-  public state = { query: '', queryProp: '', formSubmitted: false };
+      (targets[nextSectionIndex] as HTMLElement).focus();
+      // Possibly we want to scroll the current result into view in results or content?
+    },
+    [isMobile, hasSearchResults]
+  );
 
-  public render() {
-    const openMenu = (e: React.MouseEvent<HTMLButtonElement>) => {
+  useKeyCombination(searchKeyCombination, cycleSearchRegions);
+
+  return null;
+}
+
+function Topbar(props: Props) {
+  const openMenu = React.useCallback(
+    (e: React.MouseEvent<HTMLButtonElement>) => {
       e.preventDefault();
-      this.props.openMobileMenu();
-    };
+      props.openMobileMenu();
+    },
+    [props]
+  );
+  const newButtonEnabled = !!props.searchButtonColor;
+  const prevQuery = React.useRef('');
+  const [query, setQuery] = React.useState('');
+  const [formSubmitted, setFormSubmitted] = React.useState(false);
+  const state = React.useMemo(
+    () => ({query, formSubmitted}),
+    [query, formSubmitted]
+  );
+
+  if (props.query) {
+    if (props.query !== query && props.query !== prevQuery.current) {
+      setQuery(props.query);
+    }
+    prevQuery.current = props.query;
+  }
 
-    const onSearchSubmit = (e: React.FormEvent) => {
+  const onSearchChange = React.useCallback(
+    (e: React.FormEvent<HTMLInputElement>) => {
+      setQuery((e.currentTarget as any).value);
```

Comment:
Eh I get that the form will always have currentTarget, but I'd at least cast it as `{ value: string }` or whatever it is